src/notation/qml/MuseScore/NotationScene/internal/EditStyle/MeasureNumbersPage.qml

```diff
@@ -0,0 +1,428 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2021 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+import QtQuick 2.15
+import QtQuick.Layouts 1.15
+import QtQuick.Controls
+
+import MuseScore.NotationScene 1.0
+import Muse.UiComponents 1.0
+import Muse.Ui 1.0
+
+StyledFlickable {
+    id: root
+
+    signal goToTextStylePage(int index)
+
+    contentWidth: column.width
+    contentHeight: column.height
+
+    MeasureNumbersPageModel {
+        id: barNumbersModel
+    }
+
+    ColumnLayout {
+        id: column
+        spacing: 12
+
+        StyledGroupBox {
+            Layout.fillWidth: true
+            Layout.minimumWidth: 500
+            title: qsTrc("notation/editstyle/voltas", "Measure numbers")
+
+            ColumnLayout {
+                width: parent.width
+                spacing: 12
+
+                ColumnLayout {
+                    width: parent.width
+                    spacing: 8
+
+                    StyleToggle {
+                        styleItem: barNumbersModel.showMeasureNumber
+                        text: qsTrc("notation/editstyle/voltas", "Show measure numbers")
+                    }
+
+                    CheckBox {
+                        enabled: barNumbersModel.showMeasureNumber.value === true
+                        checked: barNumbersModel.showMeasureNumberOne.value === true
+                        onClicked: barNumbersModel.showMeasureNumberOne.value = !barNumbersModel.showMeasureNumberOne.value
+                        text: qsTrc("notation/editstyle/voltas", "Show initial measure number")
+                    }
+                }
+
+                ColumnLayout {
+                    enabled: barNumbersModel.showMeasureNumber.value === true
+                    width: parent.width
+                    spacing: 8
+
+                    StyledTextLabel {
+                        text: qsTrc("notation/editstyle/voltas", "Text style")
+                        horizontalAlignment: Text.AlignLeft
+                    }
+
+                    RowLayout {
+                        spacing: 4
+
+                        StyledDropdown {
```

Comment:
`StyledDropdown` is broken and opens behind the style dialog. This is due to (our favourite!) Qt widgets and QML components not playing nicely together. I think the root cause is that `StyledDropdown` doesn't implement the QML popup type.  I battled this for a long time when making the chord symbols page and eventually had to give up and create the `ComboBoxDropdown` element to use in the style dialog.
<img width="935" height="412" alt="Screenshot 2025-07-11 at 08 56 36" src="https://github.com/user-attachments/assets/6212c955-6ba9-4a96-adaf-0f3259c2e7a6" />

Reply:
Ah - good to know.

Reply:
Yes, works for me now